homeassistant/components/sensor/__init__.py

```diff
@@ -259,7 +259,9 @@ def _numeric_state_expected(self) -> bool:
             return True
         # Sensors with custom device classes are not considered numeric
         device_class = try_parse_enum(SensorDeviceClass, self.device_class)
-        return not (device_class is None or device_class in NON_NUMERIC_DEVICE_CLASSES)
+        return (
+            device_class is not None and device_class not in NON_NUMERIC_DEVICE_CLASSES
+        )
```

Comment:
```suggestion
        return return device_class not in {None, *NON_NUMERIC_DEVICE_CLASSES}
```

Alternative suggestion, thnx to @epenet ?